pkg/host-disk/host-disk_test.go

```diff
@@ -0,0 +1,227 @@
+	Describe("HostDisk with 'Disk' type", func() {
+		It("Should not create a disk.img when it exists", func() {
+			By("Creating a disk.img before adding a HostDisk volume")
+			tmpDiskImg := createTempDiskImg("volume1")
+
+			By("Creating a new minimal vmi")
+			vmi := v1.NewMinimalVMI("fake-vmi")
+
+			By("Adding a HostDisk volume for existing disk.img")
+			addHostDisk(vmi, "volume1", v1.HostDiskExists, "")
+
+			By("Executing CreateHostDisks which should not create a disk.img")
+			err := CreateHostDisks(vmi)
+			Expect(err).NotTo(HaveOccurred())
+
+			// check if disk.img has the same modification time
+			// which means that CreateHostDisks function did not create a new disk.img
+			hostDiskImg, _ := os.Stat(vmi.Spec.Volumes[0].HostDisk.Path)
+			Expect(tmpDiskImg.ModTime()).To(Equal(hostDiskImg.ModTime()))
+		})
+		It("Should not create a disk.img when it does not exist", func() {
+			By("Creating a new minimal vmi")
+			vmi := v1.NewMinimalVMI("fake-vmi")
+
+			By("Adding a HostDisk volume")
+			addHostDisk(vmi, "volume1", v1.HostDiskExists, "")
+
+			By("Executing CreateHostDisks which should not create disk.img")
+			err := CreateHostDisks(vmi)
+			Expect(err).NotTo(HaveOccurred())
+
+			// disk.img should not exist
+			_, err = os.Stat(vmi.Spec.Volumes[0].HostDisk.Path)
+			Expect(true).To(Equal(os.IsNotExist(err)))
+		})
+	})
+
+	Describe("HostDisk with 'DiskOrCreate' type", func() {
+		Context("With multiple HostDisk volumes", func() {
+			Context("With non existing disk.img", func() {
+				It("Should create disk.img if there is enough space", func() {
+					By("Creating a new minimal vmi")
+					vmi := v1.NewMinimalVMI("fake-vmi")
+
+					By("Adding a HostDisk volumes")
+					addHostDisk(vmi, "volume1", v1.HostDiskExistsOrCreate, "64Mi")
+					addHostDisk(vmi, "volume2", v1.HostDiskExistsOrCreate, "128Mi")
+					addHostDisk(vmi, "volume3", v1.HostDiskExistsOrCreate, "80Mi")
+
+					By("Executing CreateHostDisks which should create disk.img")
+					err := CreateHostDisks(vmi)
+					Expect(err).NotTo(HaveOccurred())
+
+					// check if images exist and the size is adequate to requirements
+					img1, err := os.Stat(vmi.Spec.Volumes[0].HostDisk.Path)
+					Expect(err).NotTo(HaveOccurred())
+					Expect(img1.Size()).To(Equal(int64(67108864))) // 64Mi
+
+					img2, err := os.Stat(vmi.Spec.Volumes[1].HostDisk.Path)
+					Expect(err).NotTo(HaveOccurred())
+					Expect(img2.Size()).To(Equal(int64(134217728))) // 128Mi
+
+					img3, err := os.Stat(vmi.Spec.Volumes[2].HostDisk.Path)
+					Expect(err).NotTo(HaveOccurred())
+					Expect(img3.Size()).To(Equal(int64(83886080))) // 80Mi
+				})
+				It("Should stop creating disk images if there is not enough space and should return err", func() {
+					By("Creating a new minimal vmi")
+					vmi := v1.NewMinimalVMI("fake-vmi")
+
+					By("Adding a HostDisk volumes")
+					addHostDisk(vmi, "volume1", v1.HostDiskExistsOrCreate, "64Mi")
+					addHostDisk(vmi, "volume2", v1.HostDiskExistsOrCreate, "1E")
```

Comment:
I am creating a directory in temp for sparse files and after each test, it's removed.

But talking specifically about ```1E```, I took this because it is the biggest unit from ```resource.Quantity``` (https://github.com/kubernetes/apimachinery/blob/master/pkg/api/resource/quantity.go#L46) and I'm sure that nobody has a disk like this, ofc I could take some Peta value... but ```1E``` says "oh man this is huge". So, I am able to test condition if there is enough space for a disk.img.

Even if my check would be wrong, it still shouldn't affect the system because this is a sparse file.